<launch>

  <!-- UAV namespace -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />

  <!-- Debug options -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- Nodelet manager options -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_external_wrench_estimator_manager" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">

    <!-- ExternalWrenchEstimator nodelet -->
    <node pkg="nodelet" type="nodelet" name="external_wrench_estimator" args="$(arg nodelet) external_wrench_estimator/ExternalWrenchEstimator $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Load parameters -->
      <rosparam file="$(find bump_tolerant_control)/config/external_wrench_estimator.yaml" />

      <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->
      <remap from="~motor_speed_in0" to="motor_speed/0"/>
      <remap from="~motor_speed_in1" to="motor_speed/1"/>
      <remap from="~motor_speed_in2" to="motor_speed/2"/>
      <remap from="~motor_speed_in3" to="motor_speed/3"/>
      <remap from="~motor_speed_in4" to="motor_speed/4"/>
      <remap from="~motor_speed_in5" to="motor_speed/5"/>
      <remap from="~motor_speed_in6" to="motor_speed/6"/>
      <remap from="~motor_speed_in7" to="motor_speed/7"/>
      <remap from="~odom_uav_in"     to="hw_api/odometry"/>
      <remap from="~imu_in"          to="hw_api/imu"/>


      <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->
      <remap from="~external_wrench_estimation_out"       to="external_wrench_estimator/ext_wrench" />
      <remap from="~contact_point_out"                    to="external_wrench_estimator/contact_point" /> 
      <remap from="~force_moment_norms_filt_out"          to="external_wrench_estimator/force_moment_norms_filt" />
      <remap from="~force_components_filt_out"            to="external_wrench_estimator/force_components_filt" />
      <remap from="~moment_components_filt_out"           to="external_wrench_estimator/moment_components_filt" />
      <remap from="~collision_point_marker_out"           to="external_wrench_estimator/collision_point_marker" />
      <remap from="~contact_event_out"                    to="external_wrench_estimator/contact_event" />
      <remap from="~current_controller_out"               to="external_wrench_estimator/current_controller" />

    </node>

  </group>

</launch>